### ISSUE TITLE
Subs loaded as episodes

### ISSUE THREAD

**Issue by bluboyo** [xbmc/vfs.rar#86]
New bug now when it works with the library. I use 18.7 nightly and 2.3.1 Works very nice but the subs folder is read as an epside two times. So one episide for the first rar, then the same again for the rar inside the rar. So now I have Three episodes inside kodi when there is a seperate subsfolder in rar.

**Comment by amazingdash** [comment]
This looks like the problem I mentioned here: https://github.com/xbmc/vfs.rar/issues/68#issuecomment-612691986

I just tested with 2.3.1 and I still get  `.sub`files added to my tv shows library. From the scan log:
```
VideoInfoScanner: Adding new item to tvshows:rar://rar%3a%2f%2f%252f[...]%252fSubs%252f[...].subs.rar%2f[...].rar/[...].sub
```

If I URL-decode the path _twice_, I get this:
```
rar://rar://path/to/directory/Subs/episode.subs.rar/episode.rar/episode.sub
```

Non-nested rars are fine (the .sub doesn't get added to the library).

**Comment by amazingdash** [comment]
@faltaren ugly workaround that seems to work:
```xml
<advancedsettings version="1.0">
  <video>
    <excludetvshowsfromscan>
      <regexp>\.sub$</regexp>
    <excludetvshowsfromscan>
  <video>
</advancedsettings>
```

**Comment by bluboyo** [comment]
Were do i paste this?

**Comment by amazingdash** [comment]
https://kodi.wiki/view/Advancedsettings.xml#excludetvshowsfromscan

**Comment by bluboyo** [comment]
Will try. Thx :)

**Comment by bluboyo** [comment]
Didnt work :(. Did a total rescan of the show (removed and readded) and it still scans the subs as episodes.

**Comment by Sopor** [comment]
@faltaren What is the name of the subs?

This is regex and `\.sub$` means that the file need to end with `.sub`. Maybe it is better is you exclude the subs folder.

Try use this instead `<regexp>(?i)[\\/]subs[\\/]</regexp>`

**Comment by bluboyo** [comment]
Noop. Still the same. Removed, readded. Clean. Nothing...

**Comment by bluboyo** [comment]
@AlwinEsch are you working on this? Really anoying to have 3 of every ep 😫

**Comment by AlwinEsch** [comment]
I look soon to fix it, was shortly on another project.

**Comment by bluboyo** [comment]
Thx 😊

**Comment by Sopor** [comment]
@faltaren How do you have your subtitles?
I did a test here with Kodi 18.6 and 18.7 (compiled 2020-05-02) and vfs.rar 2.3.1 and i have no problems at all.
I have a `Subs` folder where the sub files are in a RAR file.
This was movies but it is hard to find TV episodes that have external subtitles. Most of them have the subtitles in the video file.

**Comment by bluboyo** [comment]
Its folder in the episode that is named subs. Then rar and sfv file

**Comment by Sopor** [comment]
Yeah, same as me then, but i only tried it on movies and it seems to work for me.

**Comment by Sopor** [comment]
Yes, now i manage to get TV shows to create not three but at least two of every episodes. I even tried exclude `Subs` as i asked you to try, but i still get two of every episodes.

In my case both the subs files are compressed in `best` even the rar inside the rar.

**Comment by Sopor** [comment]
It seems that both these will be ignored when scanning RAR files
```
        <excludefromscan action="append">
          <regexp>(?i)[\\/]proof|subs|sample[\\/]</regexp>
        </excludefromscan>
         <excludefromlisting action="append">
          <regexp>(?i)[\\/](proof|subs|sample)[\\/]</regexp>
        </excludefromlisting>
```

**Comment by bluboyo** [comment]
You see 😋

**Comment by amazingdash** [comment]
> I look soon to fix it, was shortly on another project.

@AlwinEsch hey, sorry to bother you but have you had any time to reproduce/fix this? Kodi 18.7 has been released and it did nothing for this specific issue.

**Comment by AlwinEsch** [comment]
Think this becomes no more fixed inside Leia, there comes now 18.8 and if nothing special come, then the last release and then only becomes fixed in Kodi 19.0 Matrix.

**Comment by bluboyo** [comment]
Ok. Sounds fine 😊

**Comment by bluboyo** [comment]
If there was a fix for this in 18.8 i didnt work. Removed everything and rescanned. Still double/tripple eps.

**Comment by AlwinEsch** [comment]
See my note before:
> Think this becomes **no more** fixed inside Leia, there comes now 18.8 and if nothing special come, then the last release and then only becomes fixed in Kodi 19.0 Matrix.

On Leia was no time anymore on his last release to add, only within Matrix I can look about.

**Comment by bluboyo** [comment]
Aha så "Fixes subtitle handling from archives" is something else

**Comment by kraaa** [comment]
https://github.com/xbmc/xbmc/issues/18232 
Might have a temp solution. One guy got it fixed.
